internal/core/platform.rs

```diff
@@ -269,6 +269,13 @@ pub enum WindowEvent {
         /// ```
         text: SharedString,
     },
+    /// The window's scale factor has changed. This can happen for example when the display's resolution
+    /// changes, the user selects a new scale factor in the system settings, or the window is moved to a
+    /// different screen.
```

Comment:
Maybe mention that the backend can also call that to set the initial scale factor?